plugins/woocommerce/src/StoreApi/Utilities/QuantityLimits.php

```diff
@@ -92,51 +126,36 @@ public function limit_to_multiple( int $number, int $multiple_of, string $roundi
 	/**
 	 * Check that a given quantity is valid according to any limits in place.
 	 *
-	 * @param integer           $quantity Quantity to validate.
-	 * @param \WC_Product|array $cart_item Cart item.
+	 * @param integer $quantity Quantity to validate.
+	 * @param array   $cart_item Cart item.
 	 * @return \WP_Error|true
 	 */
 	public function validate_cart_item_quantity( $quantity, $cart_item ) {
-		$limits = $this->get_cart_item_quantity_limits( $cart_item );
+		$limits  = $this->get_cart_item_quantity_limits( $cart_item );
+		$product = $cart_item['data'] ?? false;
+
+		if ( ! $product instanceof \WC_Product ) {
+			return true;
+		}
 
 		if ( ! $limits['editable'] ) {
-			return new \WP_Error(
-				'readonly_quantity',
-				__( 'This item is already in the cart and its quantity cannot be edited', 'woocommerce' )
-			);
+			/* translators: 1: product name */
+			return new \WP_Error( 'readonly_quantity', sprintf( __( 'The quantity of &quot;%1$s&quot; cannot be changed', 'woocommerce' ), $product->get_name() ) );
 		}
 
 		if ( $quantity < $limits['minimum'] ) {
-			return new \WP_Error(
-				'invalid_quantity',
-				sprintf(
-					// Translators: %s amount.
-					__( 'The minimum quantity that can be added to the cart is %s', 'woocommerce' ),
-					$limits['minimum']
-				)
-			);
+			/* translators: 1: product name 2: minimum quantity */
+			return new \WP_Error( 'invalid_quantity', sprintf( __( 'The minimum quantity of &quot;%1$s&quot; allowed in the cart is %2$s', 'woocommerce' ), $product->get_name(), $limits['minimum'] ) );
```

Comment:
Any idea why those quotes were pre-escaped?

Reply:
No; is that an issue? Could use left/right quotes instead.